core/src/banking_stage.rs

```diff
@@ -1015,324 +1051,35 @@ impl BankingStage {
                 }
             });
 
-        if hold {
-            for deserialized_packet in buffered_packet_batches.iter_mut() {
-                deserialized_packet.forwarded = true;
-            }
-        } else {
+        if !hold {
             slot_metrics_tracker.increment_cleared_from_buffer_after_forward_count(
                 filter_forwarding_result.total_forwardable_packets as u64,
             );
             tracer_packet_stats.increment_total_cleared_from_buffer_after_forward(
                 filter_forwarding_result.total_tracer_packets_in_buffer,
             );
-            buffered_packet_batches.clear();
-        }
-    }
-
-    /// Filter out packets that fail to sanitize, or are no longer valid (could be
```

Comment:
All this code is moved (untouched) to unprocessed_transaction_storage for better encapsulation

Reply:
This could be a separate PR as well I think

Reply:
Caused some issues because we didn't have an unprocessed_transaction_storage to invoke on, we're still using packet batches on master.